app/models/chess_piece.rb

```diff
@@ -10,6 +10,16 @@ def vertical_move?(endpoint_x, endpoint_y)
     (position_y != endpoint_y) && (position_x === endpoint_x)
   end
 
+  def vertically_obstructed?(endpiont_x, endpoint_y)
+    ((position_y + 1)..endpoint_y).each do |y|
```

Comment:
`each` will always return whatever it was called on. in this case it will *always* return the range represented as an array.

take a look at http://ruby-doc.org/core-2.3.0/Enumerable.html#method-i-find
it might help you with this problem